Flaky test: 2/10k GlobalBinaryLoggingOptions

```
--- FAIL: Test (166.58s)
    --- FAIL: Test/GlobalBinaryLoggingOptions (0.07s)
        tlogger.go:116: INFO server.go:631 [core] [Server #992] Server created  (t=+502.496µs)
        tlogger.go:116: INFO clientconn.go:105 [core] [Channel #993] Channel created  (t=+809.394µs)
        tlogger.go:116: INFO clientconn.go:1547 [core] [Channel #993] original dial target is: "whatever:///127.0.0.1:46143"  (t=+934.093µs)
        tlogger.go:116: INFO server.go:819 [core] [Server #992 ListenSocket #994] ListenSocket created  (t=+1.155992ms)
        tlogger.go:116: INFO clientconn.go:1554 [core] [Channel #993] parsed dial target is: {Scheme:whatever Authority: Endpoint:127.0.0.1:46143 URL:{Scheme:whatever Opaque: User: Host: Path:/127.0.0.1:46143 RawPath: OmitHost:false ForceQuery:false RawQuery: Fragment: RawFragment:}}  (t=+2.353[183](https://github.com/grpc/grpc-go/actions/runs/3503743424/jobs/5868990699#step:8:184)ms)
        tlogger.go:116: INFO clientconn.go:263 [core] [Channel #993] Channel authority set to "127.0.0.1:46143"  (t=+2.550882ms)
        tlogger.go:116: INFO resolver_conn_wrapper.go:175 [core] [Channel #993] Resolver state updated: {
              "Addresses": [
                {
                  "Addr": "127.0.0.1:46143",
                  "ServerName": "",
                  "Attributes": null,
                  "BalancerAttributes": null,
                  "Type": 0,
                  "Metadata": null
                }
              ],
              "ServiceConfig": null,
              "Attributes": null
            } (resolver returned new addresses)  (t=+2.877579ms)
        tlogger.go:116: INFO balancer_conn_wrappers.go:274 [core] [Channel #993] Channel switches to new LB policy "pick_first"  (t=+3.134477ms)
        tlogger.go:116: INFO clientconn.go:725 [core] [Channel #993 SubChannel #995] Subchannel created  (t=+3.400075ms)
        tlogger.go:116: INFO clientconn.go:1100 [core] [Channel #993 SubChannel #995] Subchannel Connectivity change to CONNECTING  (t=+3.676273ms)
        tlogger.go:116: INFO clientconn.go:1211 [core] [Channel #993 SubChannel #995] Subchannel picks a new address "127.0.0.1:46143" to connect  (t=+3.872572ms)
        tlogger.go:116: INFO clientconn.go:418 [core] [Channel #993] Channel Connectivity change to CONNECTING  (t=+4.415168ms)
        tlogger.go:116: INFO clientconn.go:1100 [core] [Channel #993 SubChannel #995] Subchannel Connectivity change to READY  (t=+5.389861ms)
        tlogger.go:116: INFO clientconn.go:418 [core] [Channel #993] Channel Connectivity change to READY  (t=+5.661659ms)
        end2end_test.go:8266: want 8 server side binary logging events, got 7
        tlogger.go:116: INFO clientconn.go:418 [core] [Channel #993] Channel Connectivity change to SHUTDOWN  (t=+11.934114ms)
        tlogger.go:116: INFO clientconn.go:1100 [core] [Channel #993 SubChannel #995] Subchannel Connectivity change to SHUTDOWN  (t=+12.38491ms)
        tlogger.go:116: INFO clientconn.go:1425 [core] [Channel #993 SubChannel #995] Subchannel deleted  (t=+12.580909ms)
        tlogger.go:116: INFO clientconn.go:1055 [core] [Channel #993] Channel deleted  (t=+12.72[210](https://github.com/grpc/grpc-go/actions/runs/3503743424/jobs/5868990699#step:8:211)8ms)
        tlogger.go:116: INFO server.go:770 [core] [Server #992 ListenSocket #994] ListenSocket deleted  (t=+13.179005ms)
        tlogger.go:116: INFO types.go:68 [channelz] attempt to delete child with id 995 from a parent (id=993) that doesn't currently exist  (t=+14.737493ms)
        tlogger.go:116: INFO types.go:68 [channelz] attempt to delete child with id 995 from a parent (id=993) that doesn't currently exist  (t=+15.17219ms)
FAIL
FAIL	google.golang.org/grpc/test	397.167s
```

--- a/server.go
+++ b/server.go
@@ -1443,7 +1443,6 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 	// TODO: Should we be logging if writing status failed here, like above?
 	// Should the logging be in WriteStatus?  Should we ignore the WriteStatus
 	// error or allow the stats handler to see it?
-	err = t.WriteStatus(stream, statusOK)
 	if len(binlogs) != 0 {
 		st := &binarylog.ServerTrailer{
 			Trailer: stream.Trailer(),
@@ -1453,7 +1452,7 @@ func (s *Server) processUnaryRPC(t transport.ServerTransport, stream *transport.
 			binlog.Log(st)
 		}
 	}
-	return err
+	return t.WriteStatus(stream, statusOK)
 }
 
 // chainStreamServerInterceptors chains all stream server interceptors into one.
@@ -1659,7 +1658,6 @@ func (s *Server) processStreamingRPC(t transport.ServerTransport, stream *transp
 			ss.trInfo.tr.SetError()
 			ss.mu.Unlock()
 		}
-		t.WriteStatus(ss.s, appStatus)
 		if len(ss.binlogs) != 0 {
 			st := &binarylog.ServerTrailer{
 				Trailer: ss.s.Trailer(),
@@ -1669,6 +1667,7 @@ func (s *Server) processStreamingRPC(t transport.ServerTransport, stream *transp
 				binlog.Log(st)
 			}
 		}
+		t.WriteStatus(ss.s, appStatus)
 		// TODO: Should we log an error from WriteStatus here and below?
 		return appErr
 	}
@@ -1677,7 +1676,6 @@ func (s *Server) processStreamingRPC(t transport.ServerTransport, stream *transp
 		ss.trInfo.tr.LazyLog(stringer("OK"), false)
 		ss.mu.Unlock()
 	}
-	err = t.WriteStatus(ss.s, statusOK)
 	if len(ss.binlogs) != 0 {
 		st := &binarylog.ServerTrailer{
 			Trailer: ss.s.Trailer(),
@@ -1687,7 +1685,7 @@ func (s *Server) processStreamingRPC(t transport.ServerTransport, stream *transp
 			binlog.Log(st)
 		}
 	}
-	return err
+	return t.WriteStatus(ss.s, statusOK)
 }
 
 func (s *Server) handleStream(t transport.ServerTransport, stream *transport.Stream, trInfo *traceInfo) {

